file_store/src/file_sink.rs

```diff
@@ -363,24 +360,28 @@ impl FileSink {
                     let prev_path = active_sink.path.clone();
                     self.deposit_sink(&prev_path).await?;
                     self.active_sink = Some(self.new_sink().await?);
+                } else {
```

Comment:
This is still not correct.. if active sink size + buf_len is more than max_size it's creating a new sink but then you're not writing the data to the new sink, leading to a lost buffer

Reply:
4f82d9a  addresses this particular concern I think. We basically flush the current active sink which is full. And then create a new one, send the incoming buf and attach it. Regardless, this full patch may still not be correct. More to dig through.